packages/@lwc/engine/src/framework/hooks.ts

```diff
@@ -186,10 +173,10 @@ export function allocateChildrenHook(vnode: VCustomElement) {
 
 export function createViewModelHook(vnode: VCustomElement) {
     const elm = vnode.elm as HTMLElement;
-    if (!isUndefined(getAssociatedVMIfPresent(elm))) {
-        // There is a possibility that a custom element is registered under tagName,
-        // in which case, the initialization is already carry on, and there is nothing else
-        // to do here since this hook is called right after invoking `document.createElement`.
+    if (!isUpgradableElement(elm)) {
+        // Someone else claimed this custom element,
+        // most likely a native web component or an LWC
+        // component registered as a web component.
```

Comment:
I will argue the inverse. Today, it breaks badly, because the browser will upgrade the element with the defined custom element before we get to this point, and we are being foolish, and do all kind of things with that element even though the element belongs to someone else, the result is completely unpredictable. Imagine that the CE has argument observation, or listeners defined, etc. all that will continue to work, but on top of an element whos proto chain has changed drastically.

With this change, if the element is claimed, then that's the end of the story for LWC, it will skip everything else.

Reply:
@caridy Is this not too late to eject from the creation? 
If the custom element creation has reached this point, it means that we have identified an element in the template as a custom element and we have activated the `CustomElementHook.create` routine. This check is only shifting the point of failure to the next state of `CustomElementHook.create` as they expect the element to be associated with a vm.